test/sampleapp/config.yaml

```diff
@@ -43,10 +45,12 @@ languages:
       - "Dockerfile"
   - language: "ruby"
     expectedOutput: "Hello Ruby Sample v1!"
+    preCommands:
+      - exec: "cp"
+        args: "../../docs/serving/samples/hello-world/helloworld-ruby/Gemfile.lock helloworld-ruby_tmp/Gemfile.lock"
```

Comment:
If the intention is to have custom cp statements for every file which is not inlined, there is more, for example, node.js has `package-lock.json`. A language-by-language review will be necessary.

Reply:
done

Reply:
Did a language-by-language review and that's already the last piece missing